### ISSUE TITLE
containsPoint is not float save

### ISSUE THREAD

**Issue by Fuzzyma** [tmpvar/circle2#6]
Duo to the use of `squaredLength` the precision of both vectors get multiplied and thereby halfed.
To avoid rounding errors when comparing the length and the radius one should use the normal and not the squared number and additionally `Vec2.clean()` the distance from both vectors.

So instead of
```javascript
return this.position.subtract(vec, true).lengthSquared() <= this._radiusSquared;
```
it should be
```javascript
return Vec2.clean(this.position.distance(vec)) <= this.radius()
```

While an imprecise normal radius could be `499.99999999945` the square is already `249999.99999945` which is not covered from `Vec2.clean()` anymore.

**Comment by tmpvar** [comment]
Yeah, you are right that this is not consistently handled right now.  I'd argue that no matter how far we take this, there will always be a side case as long as we're using floating point.  I have half a mind to add a "rational number" mode to vec2

**Comment by Fuzzyma** [comment]
I think you can take it a long way and prevent many errors when you care about the savety a number has. E.g. when you make the numbers save till a precision of 8, every comparison should be done with a precision of 7. That way it should be ensured, that everything works correctly. Even if erros remain you still fixed nearly all other.
I came across this error, when I intersected 2 circles and afterwards checked if the intersection points are in the circle (they are exactly on the border so...). I think this is the extreme edgecase which would be covered with the approach above.

I dont know how good rational numbers would work. You would still have input values à la `new Vec2(1.3453651234522,2.32413412321`) and you would have to break down these to a rational number. Also what about irrational numbers (Pi, sqrt...)?
